### DISCUSSION TITLE
Schort Circuit with PandaPower

### DISCUSSION THREAD

**Discussion by juanmguardiola** [e2nIEE/pandapower#1058] [Q&A]
Hi, I am new using pandapower and I have a few questions regarding the short circuit module as I am using it to perform a protection coordination study:

1.  Is there is any way to perform a SC fault at each node (independently) and  to obtain the short circuit current values through each line as in Power Factory? (not the maximum or minimum current).

2.  Is it possible to get the short circuit current seen by a line switch?

**Comment by wangzhenassd** [comment]
Hi, to question1: If you use the calc_sc and set return_all_currents then you can get the SC-current of all lines with the fault on each node.
to question 2: I think not, currently it is only possible to calculate sc current on bus and branches not on switches

**Reply by lthurner** [reply]
Correct. The branch currents are discussed in detail in #862

**Reply by juanmguardiola** [reply]
Thanks! I will give it a try and tell you how it goes.

**Comment by lthurner** [comment]
About 2: shouldn't the short-circuit seen by a line switch be exactly the same as the one seen by the line?

**Reply by juanmguardiola** [reply]
That is correct, they are the same. I wanted to get them for practical purposes and to save a few lines and comparissons.

**Reply by lthurner** [reply]
Could be an interesting idea to just map this internally to the switches and put it in a res_switch_sc. Wouldn't require a change in the short circuit calculation, just some internal data mapping. If you are doing this anyway, you could try to submit a PR on this?
